scripts/addon/build.py

```diff
@@ -438,6 +438,11 @@ def qtquery(qmake, propname):
                 xlifftool_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), "..", "utils", "xlifftool.py")
                 xlifftool_cmd = [sys.executable, xlifftool_path, "-C", f"--locale={locale}", xliff_path]
                 xlifftool = subprocess.run(xlifftool_cmd, capture_output=True)
+                # This completeness metric can be out-of-date, sometimes reporting a higher-than-actual completeness after a new string is
+                # added to the app. This happens because the .xliff file doesn't have all the source strings, so an incorrect denominator is used.
+                # To show up in the .xliff, there must be a complete cycle of a new string being pulled into Pontoon (via the `extract new strings`
```

Comment:
It seems a timing issue. Does this happen with new strings only?

Does it mean that the release cycle is faster than the pontoon string merging task? How can it be?

I would prefer to detect this and throw errors at build time.

Reply:
Yes, it's only for new strings, and only on languages that were previously 100% translated. It can mean the testing/RC is moving quicker than the strings do, as they can take up to 5 days or so - there are two PRs that need to happen to get everything working on our side:
1. We add strings into our codebase (in this case, Sept 29)
2. [Extract new strings and update all locales](https://github.com/mozilla-l10n/mozilla-vpn-client-l10n/commit/9c0b512c2ea0f5710133c72cf3a07210381e2a3e) job is run, and merged into the translations repo (happened Oct 2 in this case, it’s set to run Mon-Thurs at 7am)
3. We update our codebase with the new translations (in this situation, [Oct 3rd](https://github.com/mozilla-mobile/mozilla-vpn-client/commit/bc47ff046c4a436ad8be7a6bf23690a2fee86d36))

One alternative I considered is changing how we count translation strings, which would fix the issue of miscalculated translation completeness. We'd have to write a new script for this, which seemed more code, more fragile, and more time consuming than always falling back to English.

Another alternative I considered is to run the extract new strings job on our repo before building (while using the same method of counting strings) - Essentially duplicate the Pontoon repo's ["Extract new strings" job](https://github.com/mozilla-l10n/mozilla-vpn-client-l10n/actions/runs/6530122985/workflow#L54), which runs scripts that live in our repo. This would put the sources in the `.xilff` files, preventing the bug where we misreport languages as 100% translated, causing this bug.

I didn't take this route for a couple reasons:
- It seems fragile. It needs to keep in sync w/ the GH action in the other repo, and tweaks to the script could break this or the other one. I don't expect it to fail regularly, but there is a brittleness that means I'd anticipate some issue over the next many years.
- Always falling back to English is more defensive code - it means this bug will never crop up, and could protect against other issues.

Are there downsides to always falling back to English? If there aren't any, I think my preference is to not add more build complexity.